java-frontend/src/main/java/org/sonar/java/AnalysisError.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+public class AnalysisError {
+
+  private final String message;
+  private final String cause;
+  private final String filename;
+  private final String type;
```

Comment:
I would expect something a bit more robust/testable than a `String` for the type of analysis error. It would allow to be exhaustive, adding new type when we want to handle them.

if you plan to modify that for the next error, feel free to ignore it.

Reply:
For the time being it is hardcoded string but I have in mind to probably make it an enum.